test/util/server.go

```diff
@@ -323,13 +324,30 @@ func StartTestMaster() (*configapi.MasterConfig, string, error) {
 	return master, adminKubeConfigFile, err
 }
 
+func serviceAccountSecretsExist(client *kclient.Client, namespace string, sa *kapi.ServiceAccount) bool {
+	for _, secret := range sa.Secrets {
+		ns := namespace
+		if len(secret.Namespace) > 0 {
+			ns = secret.Namespace
+		}
+		_, err := client.Secrets(ns).Get(secret.Name)
```

Comment:
Check to make sure the secret is of type `SecretTypeServiceAccountToken` and we can take this patch.  I'll go do the upstream work to add a helper method to find and validate the service account token secret.